### PR TITLE
[SPARK-7556][ML][Doc] Add user guide for spark.ml Binarizer, including Scala, Java and Python examples

### PR DESCRIPTION
**Pull request by viirya**
JIRA: https://issues.apache.org/jira/browse/SPARK-7556

### CODE REVIEW

File: docs/ml-features.md [hunk 1]
```diff
@@ -183,6 +183,83 @@ for words_label in wordsDataFrame.select("words", "label").take(3):
 </div>
 </div>
 
+## Binarizer
+
+Binarization is the process of thresholding numerical features to binary features. As some probabilistic estimators make assumption that the input data is distributed according to [Bernoulli distribution](http://en.wikipedia.org/wiki/Bernoulli_distribution), a binarizer is useful for pre-processing the input data with continuous numerical features.
+
+A simple [Binarizer](api/scala/index.html#org.apache.spark.ml.feature.Binarizer) class provides this functionality. Besides the common parameters of `InputCol` and `OutputCol`, `Binarizer` has the parameter `Threshold` used for binarizing continuous numerical features. The features greater than the threshold, will be binarized to 1.0. The features equal to or less than the threshold, will be binarized to 0.0. The example below shows how to binarize numerical features.
```
Comment (jkbradley) [inline docs/ml-features.md]: Parameter names should use camelCase (as they do in the actual API).  E.g. ```inputCol```

File: docs/ml-features.md [hunk 2]
```diff
@@ -183,6 +183,83 @@ for words_label in wordsDataFrame.select("words", "label").take(3):
...
+
+<div class="codetabs">
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+import org.apache.spark.ml.feature.Binarizer
```
Comment (jkbradley) [inline docs/ml-features.md]: Need to import DataFrame

File: docs/ml-features.md [hunk 3]
```diff
@@ -183,6 +183,83 @@ for words_label in wordsDataFrame.select("words", "label").take(3):
...
+  (2, 0.2)
+)
+val dataFrame: DataFrame = sqlContext.createDataFrame(data).toDF("label", "feature")
+
+val binarizer: Binarizer = new Binarizer()
+  .setInputCol("feature")
+  .setOutputCol("binarized_feature")
+  .setThreshold(0.5)
+
+binarizer.transform(dataFrame).select("binarized_feature").collect().foreach(println)
```
Comment (jkbradley) [inline docs/ml-features.md]: Can you please split this into 2 lines, 1 creating a new DataFrame + 1 selecting and printing data? (Same for other example code.)  This should help make it easier to understand for new users.

File: docs/ml-features.md [hunk 4]
```diff
@@ -183,6 +183,83 @@ for words_label in wordsDataFrame.select("words", "label").take(3):
...
+  RowFactory.create(0, 0.1),
+  RowFactory.create(1, 0.8),
+  RowFactory.create(2, 0.2)
+));
+StructType schema = new StructType(new StructField[]{
+  new StructField("label", DataTypes.DoubleType, false, Metadata.empty()),
+  new StructField("feature", DataTypes.DoubleType, false, Metadata.empty())
+});
+DataFrame continuousDataFrame = jsql.createDataFrame(jrdd, schema);
+Binarizer binarizer = new Binarizer().setInputCol("feature")
```
Comment (jkbradley) [inline docs/ml-features.md]: Please put 1 setter call per line (as you did in Scala)

**Comment by jkbradley** [comment]
@viirya Thank you for the quick PR!  I just had a few comments.  Also, to confirm, have you run each of the examples?

Also, can you please add a little more info to the PR description since it will become the commit message?  E.g., "Added user guide example for spark.ml Binarizer, including Scala,Java,Python examples"

**Comment by viirya** [comment]
@jkbradley Yes, I have run all the examples.

File: docs/ml-features.md [hunk 5]
```diff
@@ -183,6 +183,90 @@ for words_label in wordsDataFrame.select("words", "label").take(3):
...
+continuousDataFrame = sqlContext.createDataFrame([
+  (0, 0.1),
+  (1, 0.8),
+  (2, 0.2)
+], ["label", "feature"])
+binarizer = Binarizer(threshold=0.5, inputCol="feature", outputCol="binarized_feature")
+binarizedDataFrame = binarizer.transform(continuousDataFrame)
+binarizedFeatures = binarizedDataFrame.select("binarized_feature")
+for binarized_feature in binarizedFeatures.collect():
+  print binarized_feature
```
Comment (davies) [inline docs/ml-features.md]: binarized_feature is an `Row` object, you could do like this:
```
for binarized_feature, in binarizedFeatures.collect():
  print binarized_feature
```

Comment (viirya) [inline docs/ml-features.md]: Good point. Thanks!

**Comment by jkbradley** [comment]
LGTM, merging into master and branch-1.4
@viirya Thanks!

**Pull request closed without merging**